.ci-scripts/release/announce-a-release.bash

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+# Announces a release after artifacts have been built:
+#
+# - Publishes release notes to GitHub
+# - Announces in the #announce stream of Zulip
+# - Adds a note about the release to LWIP
+#
+# Tools required in the environment that runs this:
+#
+# - bash
+# - changelog-tool
+# - curl
+# - git
+# - jq
+
+set -o errexit
+set -o nounset
+
+# Verify ENV is set up correctly
+# We validate all that need to be set in case, in an absolute emergency,
+# we need to run this by hand. Otherwise the GitHub actions environment should
+# provide all of these if properly configured
+if [[ -z "${GITHUB_ACTOR}" ]]; then
+  echo -e "\e[31mName of the user to make changes to repo as need to be set in GITHUB_ACTOR. Exiting."
+  exit 1
+fi
+
+if [[ -z "${GITHUB_TOKEN}" ]]; then
+  echo -e "\e[31mA personal access token needs to be set in GITHUB_TOKEN. Exiting."
+  exit 1
+fi
+
+if [[ -z "${GITHUB_REF}" ]]; then
+  echo -e "\e[31mThe release tag needs to be set in GITHUB_REF."
+  echo -e "\e[31mThe tag should be in the following GitHub specific form:"
+  echo -e "\e[31m    /refs/tags/announce-X.Y.Z"
+  echo -e "\e[31mwhere X.Y.Z is the version we are announcing"
+  echo -e "\e[31mExiting."
+  exit 1
+fi
+
+if [[ -z "${GITHUB_REPOSITORY}" ]]; then
+  echo -e "\e[31mName of this repository needs to be set in GITHUB_REPOSITORY."
+  echo -e "\e[31mShould be in the form OWNER/REPO, for example:"
+  echo -e "\e[31m     ponylang/ponyup"
+  echo -e "\e[31mExiting."
+  exit 1
+fi
+
+if [[ -z "${ZULIP_TOKEN}" ]]; then
+  echo -e "\e[31mA Zulip access token needs to be set in ZULIP_TOKEN. Exiting."
+  exit 1
+fi
+
+# Set up .netrc file with GitHub credentials
+cat <<- EOF > $HOME/.netrc
+      machine github.com
+      login $GITHUB_ACTOR
+      password $GITHUB_TOKEN
+      machine api.github.com
+      login $GITHUB_ACTOR
+      password $GITHUB_TOKEN
+EOF
+
+chmod 600 $HOME/.netrc
+
+git config --global user.name 'Ponylang Main Bot'
+git config --global user.email 'ponylang.main@gmail.com'
+git config --global push.default simple
+
+# Extract version from tag reference
+# Tag ref version: "announce-1.0.0"
+# Version: "1.0.0"
+VERSION="${GITHUB_REF/refs\/tags\/announce-/}"
+
+# Prepare release notes
+echo -e "\e[34mPreparing to update GitHub release notes..."
+body=$(changelog-tool get "${VERSION}")
+
+jsontemplate="
+{
+  \"tag_name\":\$version,
+  \"name\":\$version,
+  \"body\":\$body
+}
+"
+
+json=$(jq -n \
+--arg version "$VERSION" \
+--arg body "$body" \
+"${jsontemplate}")
+
+# Upload release notes
+echo -e "\e[34mUploading release notes..."
+result=$(curl -X POST "https://api.github.com/repos/${GITHUB_REPOSITORY}/releases" \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -u "${GITHUB_ACTOR}:${GITHUB_TOKEN}" \
+  --data "${json}")
+
+rslt_scan=$(echo "${result}" | jq -r '.id')
+if [ "$rslt_scan" != null ]
+then
```

Comment:
These `then`s should be moved to the previous line to be consistent with the rest of the shell scripts (2 others below)

Reply:
Fixed.